### DISCUSSION TITLE
GeneActivity promoter

### DISCUSSION THREAD

**Discussion by summer-tao** [stuart-lab/signac#533] [Q&A] [answered]
Hello, I want to annotate my scATAC-seq cluster by gene promoter accessibility.  I generate "RNA" assay using GeneActivity function, which includes the extend.upstream = 2000(promoter), but how could I visualise only the gene promoter region, please? 
What I want to generate is similar to this picture.
Fig1A
https://assets.ctfassets.net/an68im79xiti/25vjVwfBsmoeKhzMkV7ayy/5d43e75c6844c7e6f560a6e691334638/CG000234_TechnicalNote_CellTypeAnnotationUsingATAC_Rev_A.pdf

Best,
Tao.

**Comment by wudustan** [comment, marked as answer]
Dear Tao,

You can change the distance from TSS you'd like to include.

Alternatively, I had a similar query for using `RunChromVar()` to get promoter binding deviations. It's included in discussion [here](https://github.com/timoast/signac/discussions/436). I'm happy to share code.

In terms of visualisation:

```
DefaultAssay(seurat.object) <- "RNA"
FeaturePlot(object = object.seurat, 
                   pt.size = 0.7,
                   features = "YOUR GENE OF INTEREST",
                   min.cutoff = "q10",
                   max.cutoff = "q95",
                   order = TRUE)
```

Alternatively, you can calculate a score based on a number of genes and put your vector of values in `features =` instead.

**Reply by summer-tao** [reply]
Dear wudustan,

Looks super great, I will try. Thank you so much!

Best wishes,
Tao.

**Reply by summer-tao** [reply]
Dear Wudustan,

Thanks very much for your nice sharing. 
I have no clue of how to generate the nice promoter motif.matrix.
Could you please share with me how to generate the promoter motif.matrix, please?  or just tell me some key words.
I am also on my own way to solve this problem.

Best regards,
Tao.

**Reply by summer-tao** [reply]
> In terms of visualisation:
> 
> ```
> DefaultAssay(seurat.object) <- "RNA"
> FeaturePlot(object = object.seurat, 
>                    pt.size = 0.7,
>                    features = "YOUR GENE OF INTEREST",
>                    min.cutoff = "q10",
>                    max.cutoff = "q95",
>                    order = TRUE)

Dear @wudustan , Thank you very much. With the method you provided, I have successfully generated my interested gene promoter assay.
Just want to consult you, maybe I have a misunderstanding. The visualisation you mentioned above, what I understand is the gene body + promoter feature plot, because the default assay is "RNA", is it right?

Best wishes,
Tao.

**Reply by wudustan** [reply]
@summer-tao The above is a reference to the Signac vignette [Analyzing PBMC scATAC-seq](https://satijalab.org/signac/articles/pbmc_vignette.html)

Specifically where it gives you an example of how to make Gene Activity matrices:

```
gene.activities <- GeneActivity(pbmc)
# add the gene activity matrix to the Seurat object as a new assay and normalize it
pbmc[['RNA']] <- CreateAssayObject(counts = gene.activities)
pbmc <- NormalizeData(
  object = pbmc,
  assay = 'RNA',
  normalization.method = 'LogNormalize',
  scale.factor = median(pbmc$nCount_RNA)
)

DefaultAssay(pbmc) <- 'RNA'
FeaturePlot(
  object = pbmc,
  features = c('MS4A1', 'CD3D', 'LEF1', 'NKG7', 'TREM1', 'LYZ'),
  pt.size = 0.1,
  max.cutoff = 'q95',
  ncol = 3
)
```

In `FeaturePlot()` unless you specifically specify which assay to look in for your features, it will start by looking in your `DefaultAssay()`. If it can't find them in the default assay it will look in the other assays. In the `RunChromVar()` example I linked you to in the other post, the `DefaultAssay()` would be 'promoter'.

It's useful to distinguish if, for example, you have Gene Activity scores *and* chromVar promoter score assays in your object. Otherwise, use whichever assay contains the information you'd like to plot.

**Reply by summer-tao** [reply]
Hi @wudustan, Thank you very much. I will try.
